src/cljs/athens/electron.cljs

```diff
@@ -214,161 +216,122 @@
+                               dbpath)]
+        (if duplicate?
+          {:dispatch [:alert/js (str "Database already in the list")]}
           (let [dbname  (get-db-name dbpath)
                 newdb   {:path dbpath
                          :name dbname}
-                all-dbs (conj
-                          current-db-list
-                          newdb)]
+                all-dbs (conj current-db-list newdb)]
             {:db       (assoc db :db-picker/all-dbs all-dbs)
-             :dispatch [:local-storage/set-db-picker-list]})
-          (js/alert (str "Database already in the list"))))))
+             :dispatch [:local-storage/set-db-picker-list]})))))
+
 
   (reg-event-fx
       :local-storage/set-db-picker-list
       (fn [{:keys [db]} _]
-        "Save current db-picker list to local storage.
-         Why using `pr-str`? Checkout this link for
-         discussion on how to serialize data to store"
+        "Save current db-picker list to local storage. Why using `pr-str`? Checkout
+         this link for discussion on how to serialize data to store"
         ;; https://stackoverflow.com/questions/67821181/how-to-set-and-get-a-vector-in-local-storage
         (let [current-db-list (:db-picker/all-dbs db)]
-          {:local-storage/set! [ "db-picker/all-dbs"
-                                (pr-str current-db-list)]})))
+          {:local-storage/set! ["db-picker/all-dbs" (pr-str current-db-list)]})))
 
   (reg-event-fx
     :local-storage/check-db-picker-list
     (fn [{:keys [db]} _]
       "Check if local storage contains db-picker list.
-       If not it means this is the first time opening
-       Athens or local storage was cleared, if this is
-       the case we update db-picker list with the current
-       running db."
+       If not it means this is the first time opening Athens or local storage was
+       cleared, if this is the case we update db-picker list with the current running db."
       (let [val                 (cljs.reader/read-string
-                                  (js/localStorage.getItem
-                                    "db-picker/all-dbs"))
+                                  (js/localStorage.getItem "db-picker/all-dbs"))
             current-db-filepath (:db/filepath db)]
         (cond
-          (nil? val)  {:dispatch [:db-picker/add-new-db
-                                  current-db-filepath]}
+          (nil? val)  {:dispatch [:db-picker/add-new-db current-db-filepath]}
           :else       {:db (assoc db :db-picker/all-dbs val)}))))
 
   (reg-event-fx
     :db-picker/remove-db-from-list
     (fn [{:keys [db]} [_ db-path]]
-      "Remove the selected db from db-list.
-       Update local storage value with the new db list
-       Args : db-path : type str"
+      "Remove the selected db from db-list. Update local storage value with the new db list"
       (let [current-db-list (:db-picker/all-dbs db)
             new-db-list     (into [] (filter
```

Comment:
Do you know if using `vec` is better than `into []`? Genuine question

Reply:
As of clojure 1.7 depending on the type that is being converted one of them is better than the other, the data is shown [here](https://insideclojure.org/images/20150107/numbers.png)

Reply:
@tangjeff0 @sid597 Yup `vec` should be faster in most of the cases than `into` but not significantly, so unless this is in a hot path it shouldn't matter much.

`vec` is also preferred over `into` in the style guide: https://github.com/bbatsov/clojure-style-guide#converting-sequences-to-vectors

**BUT there's a catch here**: `filter` produces a `lazy-seq` which does some extra work with chunking and stuff. So if we really want performance (and we want this to be evaluated eagerly) we should use `filterv` which removes the overhead of first converting to lazy-seq chunks and directly returns a vector (fun-fact: it uses a transient vector underneath so it's even faster)

Another thing to keep in mind would be if we're combining multiple filter/maps/other operations then using a transducer would be the most effecient. It will do a single pass over the initial collection without creating additional intermediary collections.